apps/api/v2/src/modules/organizations/controllers/users/organizations-users.e2e-spec.ts

```diff
@@ -0,0 +1,274 @@
+        ...orgMembersData.map((member) => userRepositoryFixture.deleteByEmail(member.email)),
+      ]);
+      await userRepositoryFixture.deleteByEmail(user.email);
+      await organizationsRepositoryFixture.delete(org.id);
+      await app.close();
+    });
+
+    it("should be defined", () => {
+      expect(userRepositoryFixture).toBeDefined();
+      expect(organizationsRepositoryFixture).toBeDefined();
+      expect(user).toBeDefined();
+      expect(org).toBeDefined();
+    });
+
+    it("should get all org users", async () => {
+      const { body } = await request(app.getHttpServer()).get(`/v2/organizations/${org.id}/users`);
+
+      const userData = body.data.users;
+
+      expect(body.status).toBe(SUCCESS_STATUS);
+      expect(userData.length).toBe(4);
+
+      expect(userData.filter((user: { email: string }) => user.email === nonMemberEmail).length).toBe(0);
+    });
+
+    it("should only get users with the specified email", async () => {
+      const { body } = await request(app.getHttpServer())
+        .get(`/v2/organizations/${org.id}/users`)
+        .send({
+          email: userEmail,
+        })
+        .set("Content-Type", "application/json")
+        .set("Accept", "application/json");
+
+      const userData = body.data.users;
+
+      expect(body.status).toBe(SUCCESS_STATUS);
+      expect(userData.length).toBe(1);
+
+      expect(userData.filter((user: { email: string }) => user.email === userEmail).length).toBe(1);
+    });
+
+    it("should get users within the specified emails array", async () => {
+      const orgMemberEmail = orgMembersData[0].email;
+
+      const { body } = await request(app.getHttpServer())
+        .get(`/v2/organizations/${org.id}/users`)
+        .send({
+          email: [userEmail, orgMemberEmail],
+        })
+        .set("Content-Type", "application/json")
+        .set("Accept", "application/json");
+
+      const userData = body.data.users;
+
+      expect(body.status).toBe(SUCCESS_STATUS);
+      expect(userData.length).toBe(2);
+
+      expect(userData.filter((user: { email: string }) => user.email === userEmail).length).toBe(1);
+      expect(userData.filter((user: { email: string }) => user.email === orgMemberEmail).length).toBe(1);
+    });
+
+    it("should update an org user", async () => {
+      console.log("🚀 ~ it ~ userData:", user);
+
+      const { body } = await request(app.getHttpServer())
+        .get(`/v2/organizations/${org.id}/users/${user.id}`)
+        .send({})
+        .set("Content-Type", "application/json")
+        .set("Accept", "application/json");
+
+      const userData = body.data;
+      console.log("🚀 ~ it ~ userData:", userData);
+    });
+
+    // it("should create a new org user", async () => {
+    //   const newOrgUser = {
+    //     email: "new-member@org.com",
+    //     organizationRole: "MEMBER",
+    //     autoAccept: true,
+    //   }
+
+    //   // const emailSpy = jest.spyOn(EmailService, "sendSignupToOrganizationEmail")
```

Comment:
What's the best way to spy and mock this method? I want to ensure it's being called with the right values